src/run_environment/common.ml

```diff
@@ -46,6 +46,86 @@ module Unique_id = struct
   include Ketrew_pure.Internal_pervasives.Unique_id
 end
 
+(** 
+
+Generate unique filenames for a given set of uniquely identifying properties.
+
+The module generates names that remain (way) under 255 bytes because they
+are the most common
+{{:https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits}file-system limits}
+in 2016.
+
+*)
+module Name_file = struct
+  (* Additional safety: we check that there are no hash-duplicates
+     within a given execution of a program linking with Biokepi. *)
+  let db : (string, string list) Hashtbl.t = Hashtbl.create 42
+
+  let path ~readable_suffix ~from high_level_components =
```

Comment:
`from_path` means: take a path and derive a new similar one (i.e. same directory, and "depend" on it).

since many tools take one file and convert it to something else, it's just easier (no need to `Filename.basename`, `Filename.chop_extension`, `Filename.dirname` every time).

![screenshot 2016-09-09 14 12 15](https://cloud.githubusercontent.com/assets/617111/18397727/7ae284d2-7697-11e6-94a0-e017816b726c.png)

If the need really arises we can add `Name_file.from_paths` that takes more than one path (didn't seem that common so far).